src/neural/onnx/converter.cc

```diff
@@ -170,19 +175,48 @@ std::unique_ptr<OnnxConst> Converter::GetWeghtsConverter(
       return std::make_unique<FloatOnnxWeightsAdapter>(weights, dims, order);
     case WeightsToOnnxConverterOptions::DataType::kFloat16:
       return std::make_unique<Float16OnnxWeightsAdapter>(weights, dims, order);
+    case WeightsToOnnxConverterOptions::DataType::kBFloat16:
+      return std::make_unique<BFloat16OnnxWeightsAdapter>(weights, dims, order);
   }
   throw Exception("Data type " +
                   std::to_string(static_cast<int>(options_.data_type_)) +
                   " is not supported in weights converter");
 }
 
+std::unique_ptr<OnnxConst> Converter::GetScalarConverter(float in) {
+  switch (options_.data_type_) {
+    case WeightsToOnnxConverterOptions::DataType::kFloat32:
+      return std::make_unique<FloatOnnxConst>(FloatOnnxConst({in}, {1}));
+    case WeightsToOnnxConverterOptions::DataType::kFloat16:
+      return std::make_unique<Float16OnnxConst>(
+          Float16OnnxConst({FP32toFP16(in)}, {1}));
+    case WeightsToOnnxConverterOptions::DataType::kBFloat16:
+      return std::make_unique<BFloat16OnnxConst>(
+          BFloat16OnnxConst({FP32toBF16(in)}, {1}));
+  }
+  throw Exception("Data type " +
+                  std::to_string(static_cast<int>(options_.data_type_)) +
+                  " is not supported in scalar converter");
+}
+
 std::string Converter::MakeMish(OnnxBuilder* builder, const std::string& input,
                                 const std::string& name) {
   if (!options_.alt_mish || options_.opset < 9 ||
       options_.data_type_ !=
           WeightsToOnnxConverterOptions::DataType::kFloat32) {
     if (options_.opset >= 18) return builder->Mish(name, input);
-    auto flow = builder->Softplus(name + "/softplus", input);
+    std::string flow;
+    if (options_.fix_bf16 &&
+        options_.data_type_ ==
+            WeightsToOnnxConverterOptions::DataType::kBFloat16) {
+      flow = builder->Cast(name + "/to_float", input,
```

Comment:
Maybe have two helper functions:
```
std::string MaybeFromBf16Fix(std::string name, std::string flow) {
   if (!options_.fix_bf16 ||
        options_.data_type_ !=
            WeightsToOnnxConverterOptions::DataType::kBFloat16)) return flow;
   return builder->Cast(name + "/to_float", input, pblczero::TensorProto::FLOAT);
}
std::string MaybeToBf16Fix(std::string name, std::string flow) { // similar
```

and then write without the need for conditional:
```
flow = MaybeFromBf16Fix(name, flow);
flow = builder->Softplus(name + "/softplus", input);
flow = MaybeToBf16Fix(name, flow);
```